sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableCatalog.java

```diff
@@ -97,6 +102,26 @@ public interface TableCatalog extends CatalogPlugin {
    */
   Identifier[] listTables(String[] namespace) throws NoSuchNamespaceException;
 
+  /**
+   * List the tables in a namespace from the catalog by pattern string.
+   * <p>
+   * If the catalog supports views, this must return identifiers for only tables and not views.
+   *
+   * @param namespace a multi-part namespace
+   * @param pattern the filter pattern, only '*' and '|' are allowed as wildcards, others will
+   *                follow regular expression convention, case-insensitive match and white spaces
+   *                on both ends will be ignored
```

Comment:
do we have a doc page for the pattern string semantic? If we do we should reference it here.

Reply:
I searched the document and the `only possible relationship` is this one:
https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-like.html#parameters
<img width="910" alt="image" src="https://github.com/apache/spark/assets/15246973/357c77e8-ecdb-404e-b479-b7b7459fd06b">

Perhaps we should explain it in detail here?
(PS: The first pr that introduces `StringUtils.filterPattern` is: https://github.com/apache/spark/pull/12206)

Reply:
yea if they use the same implementation. The LIKE pattern doc does not even mention the `*`.

Reply:
Another option is to document it in the SHOW TABLES doc page.

Reply:
I have looked at the document `https://spark.apache.org/docs/latest/sql-ref-syntax-aux-show-tables.html#parameters`(SHOW TABLES doc page) and found that the parameter `regex_pattern` in it explains the `pattern`.
<img width="947" alt="image" src="https://github.com/apache/spark/assets/15246973/6349db2d-825e-4031-8f3e-4c984673f962">
Thank you very much for your reminder, Let's refer to it.